### ISSUE TITLE
Refresh the content with click on button

### ISSUE THREAD

**Issue by VanillaWulf** [moroshko/react-autosuggest#537]
I made an error message in my footer. This message renders with timeout of fake request.

renderSuggestionsContainer  ({ containerProps, children }) {
    return(
        <div {...containerProps}>
         <div className="footer">
           ServerError
           <button onClick={this.refreshContainer(this.state.value)}>Refresh</button>
         </div>
         </div>
      );
}

refreshContainer(value){
   this.loadSuggestions(value); 
 // I need somehow to render container one more time 
};

How can I refresh the request+render by onClick?

Thanks.

**Comment by Olliebaba** [comment]
I don't believe this merits an issue ticket and might be more suited for SO. 
However, to respond to your question:

(I am making some assumptions here without seeing fully how the component is being used.)

In your example, does `this.loadSuggestions(value);` trigger the same method that `onSuggestionsFetchRequested` is passed? I imagine this is what you'll want to call to trigger another ajax call.

As far as updating the `renderSuggestionsContainer`, I do not see `children` even being rendered so it's possible that it *is* updating, but it wont show the children. Try adding `{children}` to that return to see if they are being added.

I'm guessing you'll want to have the parent component's state track if the previous request failed. This way, you can display the button to re-request or not based on that:
```
renderSuggestionsContainer  ({ containerProps, children }) {
   return(
      <div {...containerProps}>
         {
            this.state.fetchFailed
               ? (
                  <div className="footer">
                     ServerError
                     <button onClick={this.refreshContainer(this.state.value)}>Refresh</button>
                </div>
               ) : (
                  {children}
               )
         }
    </div>
   );
}
```

I hope this helps!

**Comment by VanillaWulf** [comment]
Thank you for your reply.

> In your example, does this.loadSuggestions(value); trigger the same method that onSuggestionsFetchRequested is passed?

 not exactly. loadSuggestions - it is a kind of emulation of downloading request, but not the exact fetch.

> As far as updating the renderSuggestionsContainer, I do not see children even being rendered so it's possible that it is updating, but it wont show the children. Try adding {children} to that return to see if they are being added.

I render them in ok-cases.

Full component you may see here https://codepen.io/vanillawulf/pen/vjWVxL

The problem: the component renders the error message, but I don't know how to refresh the container exactly by clicking on Refresh-button.

Cause every function in component only refresh the state of component but not the rendering.

**Comment by Olliebaba** [comment]
A couple things that I see for general best-practices:
- When setting state in react, be sure to use [functional setState](https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b).
- It feels odd to call `setState` inside of another `setState`. Calculate all the values you want to set then just call `setState` once.

Based on your example, [I have made it work as I understand your requirements.](https://codepen.io/Olliebaba/pen/XqzxoL?editors=0011)

**Comment by VanillaWulf** [comment]
Thank you for your review and reply. It helpes me to find the decision.
I want that refresh-button will be dinamic, so use state for trigerring.
Working code - https://codepen.io/vanillawulf/pen/vjWVxL
The problem is solved.
